src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -39,9 +39,14 @@ public String getToken() {
     // Called from other synchronized code. Do not call into other synchronized code or you'll
     // risk deadlock
     public synchronized void setDistinctId(String distinctId) {
-        mUnseenSurveys.clear();
-        mUnseenNotifications.clear();
-        mDistinctId = distinctId;
+        if (mDistinctId == distinctId) {
```

Comment:
agree with the equals operator... as for the nulls, it looks like this is taken care of before the other methods call this one so not sure it is needed here.